misc/e2elive.py

```diff
@@ -84,7 +84,15 @@ def main():
         from botocore import UNSIGNED
         s3 = boto3.client("s3", config=Config(signature_version=UNSIGNED))
         tarpath = os.path.join(tempdir, tarname)
-        prefix = "indexer/e2e4"
+        if os.getenv("E2E_FILE_VERSION")=="submodule":
```

Comment:
There are some things to consider here:
1) I believe this approach, where we look up the commit hash, fails if we set the submodule to something that wasn't the latest commit. i.e. if there were 10 commits today, only one of them would be used for the nightly test. Also, would it ever work if we pin the submoule to rel/beta or master?
2) We could use the prefix approach, but then we would need to remember to update the E2E_FILE_VERSION each time the submodule is updated.

Reply:
right. this approach would only works with rel/nightly artifacts. I don't think we need to update E2E_FILE_VERSION. it's a flag for determining whether to use rel-nightly artifiact or use the one with the same commit hash as the submodule.

Reply:
Since this is running in a python script, you could use python-native git libraries. I already have an example of this in my parity job [here](https://github.com/algorand/indexer/blob/b197dc35b836d8cf443dab7c6510bd1be6f089d0/misc/parity/test_indexer_v_algod.py#L36-L40)